run_unittests.py

```diff
@@ -4212,9 +4216,16 @@ def test_clang_tidy(self):
         if is_osx():
             raise unittest.SkipTest('Apple ships a broken clang-tidy that chokes on -pipe.')
         testdir = os.path.join(self.unit_test_dir, '70 clang-tidy')
-        self.init(testdir, override_envvars={'CXX': 'c++'})
-        out = self.run_target('clang-tidy')
-        self.assertIn('cttest.cpp:4:20', out)
+        headerdir = os.path.join(testdir, 'dir.h')
+        try:
+            os.mkdir(headerdir)
+            self.init(testdir, override_envvars={'CXX': 'c++'})
+            out = self.run_target('clang-tidy')
+            self.assertIn('cttest.cpp:4:20', out)
+            self.assertNotIn(headerdir, out)
+        finally:
+            if os.path.exists(headerdir):
+                os.rmdir(headerdir)
```

Comment:
Tests should not modify the source directory. Create the directory and add a `dummy.dat` file that says something like "This file only exists here to get its enclosing directory into Git." and add it to Git instead.

Reply:
Updated both testcases to include static directory `dummydir.h`.